src/lib/server/query_handler.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <variant>
+#include "hyrise.hpp"
+#include "operators/abstract_operator.hpp"
+#include "postgres_protocol_handler.hpp"
+#include "storage/table.hpp"
+
+namespace opossum {
+
+// Store relevant information after pipeline execution
+struct ExecutionInformation {
+  std::shared_ptr<const Table> result_table;
+  // Since the PostgreSQL Wire Protocol requires the query type (such as SELECT, INSERT, UPDATE,...) we need to store
+  // the root operator's type.
+  OperatorType root_operator;
+  std::string execution_information;
+  std::string error;
+};
+
+// This class manages the interaction between the server and the database component. Furthermore, most of the SQL-based
+// error handling happens in this class.
+class QueryHandler {
+ public:
+  static ExecutionInformation execute_pipeline(const std::string& sql, const bool debug_note);
```

Comment:
Align with above member name. Maybe metrics, since it boils down to the sql statement metrics/timings being collected? debug seems a little misleading.